managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/NonDurableCursorImpl.java

```diff
@@ -48,39 +46,46 @@ public class NonDurableCursorImpl extends ManagedCursorImpl {
             // Start from last entry
             switch (initialPosition) {
                 case Latest:
-                    initializeCursorPosition(ledger.getLastPositionAndCounter());
+                    initializeCursorPosition(ledger.getLastPositionAndCounter(), inclusive);
                     break;
                 case Earliest:
-                    initializeCursorPosition(ledger.getFirstPositionAndCounter());
+                    initializeCursorPosition(ledger.getFirstPositionAndCounter(), inclusive);
                     break;
             }
         } else if (startCursorPosition.getLedgerId() == PositionImpl.EARLIEST.getLedgerId()) {
             // Start from invalid ledger to read from first available entry
-            recoverCursor(ledger.getPreviousPosition(ledger.getFirstPosition()));
+            recoverCursor(ledger.getFirstPosition(), inclusive);
         } else {
             // Since the cursor is positioning on the mark-delete position, we need to take 1 step back from the desired
             // read-position
-            recoverCursor(startCursorPosition);
+            recoverCursor(startCursorPosition, inclusive);
```

Comment:
Is it possible to add a new method in ledger: `getSpecifyPositionAndCounter(Position startPosition, boolean inclusive)`

and this can call: `initializeCursorPosition(ledger.getSpecifyPositionAndCounter(startCursorPosition, inclusive))`

Reply:
Looks we only support reading from Earliest/Latest position, so I don't think it's necessary to add this.